sunpy/io/__init__.py

```diff
@@ -1 +1,12 @@
-from sunpy.io._file_tools import *
+"""
+This subpackage contains a series of file readers and only for internal use.
+
+None of this code is intended to be used directly by a user and is not part of the sunpy public API
+and is not added to the API reference pages.
+"""
+from sunpy.util.exceptions import warn_deprecated
+
+warn_deprecated('The toplevel space of the io subpackage was never intended for public use and will be removed in the future.')
+from sunpy.io._file_tools import read_file, write_file
```

Comment:
I think this is wrong. 

I get the message if I do:
```python
In [3]: import sunpy.io._header
WARNING: SunpyDeprecationWarning: The toplevel space of the io subpackage was never intended for public use and will be removed in the future. [sunpy.io]
```
I should not be raising this warning if someone does import that.

Reply:
Or even:

```
import sunpy.io.special.asdf
```

As it stands, we can't merge this till I fix this.